src/json/json.cc

```diff
@@ -258,6 +258,143 @@ STATIC JsonUtilCode parseSetCmdArgs(ValkeyModuleString **argv, const int argc, S
     return JSONUTIL_SUCCESS;
 }
 
+typedef struct {
+    ValkeyModuleString *key_str;    // required
+    ValkeyModuleKey *key;
+    const char *path;               // required
+    const char *json;               // required
+    size_t json_len;
+    bool is_root_path;
+    size_t doc_size;
+} MSetCmdArgs;
+
+typedef struct {
+    size_t num_keys;
+    MSetCmdArgs *args_list;
+} MSetCmdArgsList;
+
+STATIC void FreeMSetCmdArgs(ValkeyModuleCtx *ctx, MSetCmdArgs* args_list, size_t num_keys) {
+    if (args_list == NULL) {
+        return;
+    }
+
+    for (size_t i = 0; i < num_keys; i++) {
+        if (args_list[i].key_str != NULL) {
+            ValkeyModule_FreeString(ctx, args_list[i].key_str);
+        }
+
+        if (args_list[i].key != NULL) {
+            ValkeyModule_CloseKey(args_list[i].key);
+        }
+
+        if (args_list[i].path != NULL) {
+            ValkeyModule_Free((void*)args_list[i].path);
+        }
+
+        if (args_list[i].json != NULL) {
+            ValkeyModule_Free((void*)args_list[i].json);
+        }
+    }
+}
+
+STATIC JsonUtilCode parseAndValidateMSetCmdArgs(ValkeyModuleCtx *ctx, ValkeyModuleString **argv, const int argc, MSetCmdArgsList *args) {
+    // we need 3 arguments for each key for MSET
+    if ((argc - 1) % 3) {
+        return JSONUTIL_WRONG_NUM_ARGS;
+    }
+    size_t num_keys = (argc - 1) / 3;
+    JsonUtilCode rc;
+
+    ValkeyModule_Log(ctx, "notice", "sizeof(MSetCmdArgsList)=%zu", sizeof(MSetCmdArgsList));
+    ValkeyModule_Log(ctx, "notice", "sizeof(MSetCmdArgs)=%zu", sizeof(MSetCmdArgs));
+    ValkeyModule_Log(ctx, "notice", "sizeof(num_keys)=%zu", sizeof(size_t));
+
+    memset(args, 0, sizeof(MSetCmdArgsList));
```

Comment:
Move this line to the top of the function, so that when the function returns at L303, the output param is zero-ed out.